lib/ipam/ipam_block_reader_writer.go

```diff
@@ -63,7 +64,8 @@ func (rw blockReaderWriter) getAffineBlocks(ctx context.Context, host string, ve
 			ids = append(ids, k.CIDR)
 		} else {
 			for _, pool := range pools {
-				if pool.Contains(k.CIDR.IPNet.IP) {
+				poolNet := cnet.MustParseCIDR(pool.Spec.CIDR)
```

Comment:
I think we want to avoid using `MustParseCIDR` in production code - IIRC it was a testing tool?

If it fails to parse it will panic. It shouldn't ever happen, since we validate the CIDR, but I think still best to handle this gracefully and avoid panics in libcalico.